[Core] Add scalar versions of `Vector*` `min/max/clamp/snap(ped)`

AThousandShips opened the pull request:
Convenience for a number of cases operating on single values

Added a lot of use cases, which I think should justify this addition, it simply saves a *lot* of code when you use repeated cases, and also reduces risk of copy errors

Optionally bound it to scripting

See also:
* https://github.com/godotengine/godot/pull/80223

If it gets merged first I will expose the `mini/minf/maxi/maxf` methods as well, and vice-versa
<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

AThousandShips committed:
[Core] Add scalar versions of `Vector*` `min/max/clamp/snap(ped)`

Convenience for a number of cases operating on single values

Nazarwadim commented on core/math/vector2.h:
```diff
@@ -89,10 +89,18 @@ struct _NO_DISCARD_ Vector2 {
 		return Vector2(MIN(x, p_vector2.x), MIN(y, p_vector2.y));
 	}
 
+	Vector2 minf(real_t p_scalar) const {
```
I would add `_FORCE_INLINE_` for `min*` `max*` methods in headers.  `_FORCE_INLINE_` reduces compile time because the compiler doesn't have to analyse the method to inline it.

AThousandShips replied:
As per the comment elsewhere I won't add it unless it's warranted, should be done with care, and as it's not done for the existing ones I'll not do it here either, it'll be inlined if it's appropriate anyway

Mickeon reviewed (commented):
Documentation is completely fine and consistent with the existing methods. I just am not sure if having it exposed is strictly necessary.

AThousandShips commented:
I'd say these methods are *more* generally useful than the vector versions, most of the time you want to clamp or snap to the same value, though in GDScript you *do* have the benefit of being able to do `vec.snapped(Vector2.ONE)` which you don't have in C++, but `vec.snappedf(1.0)` is still way cleaner, and especially `vec.snappedf(2.0)` is way better than `vec.snapped(Vector2.ONE * 2.0)`

raulsntos commented on modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs:
```diff
@@ -191,6 +191,23 @@ internal void Normalize()
+        /// <param name="min">The minimum allowed value.</param>
+        /// <param name="max">The maximum allowed value.</param>
+        /// <returns>The vector with all components clamped.</returns>
+        public readonly Vector2 ClampF(real_t min, real_t max)
```
I think we may want to use overloads in C# instead of a suffix.

```suggestion
        public readonly Vector2 Clamp(real_t min, real_t max)
```

AThousandShips replied:
Good point missed the `PosMod` below which uses the same, think I took this from `MathF` having some methods, will update

AThousandShips commented:
Went ahead and added the missing `snapped` methods to C# as they're exposed in GDScript, also updated the documentation for the non-integer ones to match the GDScript documentation which makes it clear it's component-wise

aaronfranke commented on modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs:
```diff
@@ -600,7 +617,7 @@ internal void Normalize()
 
         /// <summary>
-        /// Returns this vector with each component snapped to the nearest multiple of <paramref name="step"/>.
+        /// Returns a new vector with each component snapped to the nearest multiple of the corresponding component in <paramref name="step"/>.
```
The old sentence meant "this but snapped", if we are saying it's a new vector we need to reference that it is based on the old one. Of course that should be obvious, but it should still be stated in the doc text somehow. Maybe "Returns a copy of this vector [...]"?

AThousandShips replied:
This is used pretty much everywhere else, I'd say that's for a dedicated PR adding a new unified style, this just unifies the C# and GDScript docs

aaronfranke reviewed (approved):
I think this function is a good idea, it seems very common and useful.

deralmas reviewed (approved):
Finally gave a good look at this thing.

All the changes are very simple and they increase readability a lot, great job!

akien-mga commented:
Thanks!

AThousandShips commented:
Thank *you*!
